dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1036 @@
+        The geometry of the camera, including pixel positions and camera type.
+    camera_type : str
+        The type of the camera, derived from the geometry.
+    image_shape : int
+        The shape of the 2D image, based on the camera type.
+    n_pixels : int
+        The number of pixels in the camera.
+    pix_x : numpy.ndarray
+        The x-coordinates of the pixels rounded to three decimal places.
+    pix_y : numpy.ndarray
+        The y-coordinates of the pixels rounded to three decimal places.
+    x_ticks : list
+        Unique x-coordinates of the pixels.
+    y_ticks : list
+        Unique y-coordinates of the pixels.
+    internal_pad : int
+        Padding used to ensure that the camera pixels aren't affected at the edges.
+    rebinning_mult_factor : int
+        Multiplication factor used for rebinning.
+    index_matrix : numpy.ndarray or None
+        Matrix used for indexing, initialized to None.
+
+    Methods
+    -------
+    map_image(raw_vector)
+        Transform the raw 1D vector data into the 2D mapped image.
+    """
 
-class ImageMapper:
     def __init__(
         self,
-        camera_types=None,
-        pixel_positions=None,
-        mapping_method=None,
-        padding=None,
-        interpolation_image_shape=None,
-        mask_interpolation=False,
+        geometry,
+        config=None,
+        parent=None,
+        **kwargs,
     ):
+        """
+        Parameters
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent : ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
 
-        # image_shapes should be a non static field to prevent problems
+        # Default image_shapes should be a non static field to prevent problems
         # when multiple instances of ImageMapper are created
-        self.image_shapes = {
-            "LSTCam": (110, 110, 1),
-            "LSTSiPMCam": (234, 234, 1),
-            "FlashCam": (112, 112, 1),
-            "NectarCam": (110, 110, 1),
-            "SCTCam": (120, 120, 1),
-            "DigiCam": (96, 96, 1),
-            "CHEC": (48, 48, 1),
-            "ASTRICam": (56, 56, 1),
-            "VERITAS": (54, 54, 1),
-            "MAGICCam": (78, 78, 1),
-            "FACT": (90, 90, 1),
-            "HESS-I": (72, 72, 1),
-            "HESS-II": (104, 104, 1),
+        self.default_image_shapes = {
+            "LSTCam": 110,
+            "LSTSiPMCam": 234,
+            "FlashCam": 112,
+            "NectarCam": 110,
+            "SCTCam": 120,
+            "DigiCam": 96,
+            "CHEC": 48,
+            "ASTRICam": 56,
+            "VERITAS": 54,
+            "MAGICCam": 78,
+            "FACT": 92,
```

Comment:
Yes that is on purpose. I needed to add two more. Probably because we didn't de-rotated the pixel positions by default before. Those default images shapes are kind of magic numbers originated from the oversampling method, where one hexagonal pixels is divided into 4 squared pixels. In principle, those numbers can be retrieved from the `_get_grid()` function of the `OversamplingMapper` for any `ImageMapper` subclass.

Reply:
Solved